datasets/inception_subset.py

```diff
@@ -0,0 +1,174 @@
+########################################################################################################
+
+# TORCH
+import torch
+from torch.utils.data import Dataset, TensorDataset
+
+# Python Packages
+import numpy as np
+# import glob
+from datetime import datetime
+from typing import List, Type, Tuple, Iterator
+from collections import defaultdict  # Nested dictionaries in python
+
+# SKLearn
+from sklearn.model_selection import train_test_split
+from sklearn.model_selection import StratifiedKFold
+
+# PNN SPECIFIC
+from datasets.labels import labels
+from train import parameters
+
+
+class Modalities(Dataset):
+    """
+    A dataset class that lets the user decides which modalities to use.
+    """
+
+    def __init__(self, root_dir: str,
+                 exp_name: str,
+                 start_date,
+                 end_date,
+                 **plant_files_dict):
+        """
+        :param root_dir: path to the experiment directory
+        :param exp_name: the experiment we want to use
+        :param mods: modalities to be in the dataset, as dictionaries of initialization arguments
+
+        """
+        self.experiment = exp_name
+        self.num_plants = len(labels[exp_name])
+        self.modalities = defaultdict(dict)
+        self.plant_files_dict = plant_files_dict
+        self.num_plants = len(labels[self.experiment])
+        # todo: fix used plants to be more generic
+        self.used_plants = [plant for plant in range(self.num_plants)]
+        self.root_dir = root_dir
+        self.mods = parameters.used_modalities
+        self.files = []
+        self.start_date = start_date.date()
+        self.end_date = end_date.date()
+
+    def __len__(self):  # the number of samples in the dataset
+        return self.num_plants
+
+    def __get_file_date(self, file, plant, mod):
+        file_format = f"{self.root_dir}/{self.experiment}_{mod}_{plant}%Y-%m-%d_%H:%M:%S"
+        return datetime.strptime(file, file_format).date()
+
+    # TODO: this one did not work because getitem gets only one parameter (plant)
+    # @staticmethod
+    # def __plant_inception__(self, plant, mod):
+    #     tensors = []
+    #     data_list = self.plant_files_dict[mod][plant]
+    #     if len(data_list) > 0:
+    #         plant_name = f'plant_{plant}_'
+    #         print(plant_name, mod)
+    #         for file in data_list:
+    #             date_is = self.__get_file_date__(file, plant_name)  # file date
+    #             if self.start_date <= date_is <= self.end_date:
+    #                 inception_embedding = torch.load(file)
+    #                 tensors.append(inception_embedding)
+    #     plant_inception = torch.stack([tensor for tensor in tensors])
+    #
+    #     return plant_inception
+
+    def __plant_inception_one_mod(self, plant, mod):
+        tensors = []
+        data_list = self.plant_files_dict[mod][plant]
+        # if len(data_list) > 0:
+        if data_list is not None:
```

Comment:
you can also do
```suggestion
        if data_list:
```
(you can keep it the same if it looks more readable to you right now)